.tugboat/wp-config.tugboat.php

```diff
@@ -1,88 +1,10 @@
 <?php
-/**
- * The base configuration for WordPress
- *
- * The wp-config.php creation script uses this file during the installation.
- * You don't have to use the web site, you can copy this file to "wp-config.php"
- * and fill in the values.
- *
- * This file contains the following configurations:
- *
- * * Database settings
- * * Secret keys
- * * Database table prefix
- * * ABSPATH
- *
- * @link https://wordpress.org/documentation/article/editing-wp-config-php/
- *
- * @package WordPress
- */
+
 // ** Database settings - Do not change these. ** //
 define('DB_NAME', 'tugboat');
 define('DB_USER', 'tugboat');
 define('DB_PASSWORD', 'tugboat');
 define('DB_HOST', 'mysql');
 
-/** Database charset to use in creating database tables. */
-define( 'DB_CHARSET', 'utf8' );
-
-/** The database collate type. Don't change this if in doubt. */
-define( 'DB_COLLATE', '' );
-
-/**#@+
- * Authentication unique keys and salts.
- *
- * Change these to different unique phrases! You can generate these using
- * the {@link https://api.wordpress.org/secret-key/1.1/salt/ WordPress.org secret-key service}.
- *
- * You can change these at any point in time to invalidate all existing cookies.
- * This will force all users to have to log in again.
- *
- * @since 2.6.0
- */
-define( 'AUTH_KEY',         'put your unique phrase here' );
-define( 'SECURE_AUTH_KEY',  'put your unique phrase here' );
-define( 'LOGGED_IN_KEY',    'put your unique phrase here' );
-define( 'NONCE_KEY',        'put your unique phrase here' );
-define( 'AUTH_SALT',        'put your unique phrase here' );
-define( 'SECURE_AUTH_SALT', 'put your unique phrase here' );
-define( 'LOGGED_IN_SALT',   'put your unique phrase here' );
-define( 'NONCE_SALT',       'put your unique phrase here' );
-
-/**#@-*/
-
-/**
- * WordPress database table prefix.
- *
- * You can have multiple installations in one database if you give each
- * a unique prefix. Only numbers, letters, and underscores please!
- */
-$table_prefix = 'wp_';
-
-/**
- * For developers: WordPress debugging mode.
- *
- * Change this to true to enable the display of notices during development.
- * It is strongly recommended that plugin and theme developers use WP_DEBUG
- * in their development environments.
- *
- * For information on other constants that can be used for debugging,
- * visit the documentation.
- *
- * @link https://wordpress.org/documentation/article/debugging-in-wordpress/
- */
-define( 'WP_DEBUG', false );
-
-/* Add any custom values between this line and the "stop editing" line. */
-
-
-
-/* That's all, stop editing! Happy publishing. */
-
-/** Absolute path to the WordPress directory. */
-if ( ! defined( 'ABSPATH' ) ) {
-	define( 'ABSPATH', __DIR__ . '/' );
-}
-
-/** Sets up WordPress vars and included files. */
-require_once ABSPATH . 'wp-settings.php';
+define( 'WP_HOME', 'https://' . getenv('TUGBOAT_SERVICE_URL_HOST') );
+define( 'WP_SITEURL', 'https://' . getenv('TUGBOAT_SERVICE_URL_HOST') . '/wp' );
```

Comment:
I'm not sure what these variables do, but should the `WP_SITEURL` be set to the `/docroot` subdirectory where the code resides?

Reply:
These constants force WordPress to use the given values for the site's domain name and path to WP Core.

Context: By default, when you add links and images to a Page in WordPress, the editor embeds the _entire_ URL for the link/image, including the domain name.

Example: Say you're working locally and pull down the live site database. If WordPress is configured in the default way described above, then your database will be full of links and assets that are pointed at the live site domain.

This is why so many dev WP workflows involve the search-replace cli command.

```
wp search-replace 'https://livesite.com' 'https://local.dev'
```

But, you don't have to deal with this problem if you setup your WordPress site correctly from the beginning by defining these constants in an evironment-dynamic way.

With these constants defined, WordPress will **Not** insert the site's domain into links and assets. It will instead use relative paths without any domain.

-----

The way I have it setup here, each tugboat instance will set its domain name based on the tugboat environment variable. So each site knows its domain, and we don't have to do a search-replace on the database after each restore.